src/content/dataSubmissions/DataSubmissionsListView.tsx

```diff
@@ -245,76 +217,163 @@ const ListingView: FC = () => {
+          orderBy,
+          organization: organization?._id ?? "All",
+          status: statusFilter,
+        },
+        context: { clientName: "backend" },
+        fetchPolicy: "no-cache",
+      });
+      if (error || !d?.listSubmissions) {
+        throw new Error("Unable to retrieve submission quality control results.");
+      }
+      setData(d.listSubmissions.submissions);
+      setTotalData(d.listSubmissions.total);
+    } catch (err) {
+      setError(true);
+    } finally {
+      setLoading(false);
+    }
   };
 
-  const handleChangeRowsPerPage = (event) => {
-    setPerPage(parseInt(event.target.value, 10));
-    setPage(0);
+  const isValidOrg = (orgId: string) =>
+    orgId && !!activeOrganizations?.find((org) => org._id === orgId);
+  const isStatusFilterOption = (status: string): status is FilterForm["status"] =>
+    ["All", ...statusValues].includes(status);
+
+  const handleOrganizationChange = (organizationId: string) => {
+    if (organizationId === orgFilter) {
+      return;
+    }
+
+    if (!canChangeOrgs && isValidOrg(user?.organization?.orgID)) {
+      setValue("organization", user.organization.orgID);
+    } else if (canChangeOrgs && isValidOrg(organizationId)) {
+      setValue("organization", organizationId);
+    }
+  };
+
+  const handleStatusChange = (status: string) => {
+    if (status === statusFilter) {
+      return;
+    }
+
+    if (isStatusFilterOption(status)) {
+      setValue("status", status);
+    }
   };
 
-  const handleOnCreateSubmission = () => {
-    refetch();
+  useEffect(() => {
+    if (!activeOrganizations?.length) {
+      return;
+    }
+
+    const organizationId = searchParams.get("organization");
+    const status = searchParams.get("status");
+
+    handleStatusChange(status);
+    handleOrganizationChange(organizationId);
+  }, [
+    activeOrganizations,
+    canChangeOrgs,
+    searchParams.get("organization"),
+    searchParams.get("status"),
+  ]);
+
+  useEffect(() => {
+    if (!touchedFilters.organization && !touchedFilters.status) {
+      return;
+    }
+
+    if (canChangeOrgs && orgFilter && orgFilter !== "All") {
+      searchParams.set("organization", orgFilter);
+    } else if (orgFilter === "All") {
+      searchParams.delete("organization");
+    }
+    if (statusFilter && statusFilter !== "All") {
+      searchParams.set("status", statusFilter);
+    } else if (statusFilter === "All") {
+      searchParams.delete("status");
+    }
+
+    setTablePage(0);
+    setSearchParams(searchParams);
+  }, [orgFilter, statusFilter, touchedFilters]);
+
+  const setTablePage = (page: number) => {
+    tableRef.current?.setPage(page, true);
+  };
+
+  const handleOnCreateSubmission = async () => {
+    if (!activeOrganizations?.length) {
+      return;
+    }
+
+    try {
+      setLoading(true);
+
+      const { data: d } = await refetch();
+      if (error || !d?.listSubmissions) {
+        throw new Error("Unable to retrieve submission quality control results.");
```

Comment:
Could we update this error message to something else? I don't believe it's actually used anywhere, but if it ever is, this would be an issue.

Reply:
Oops copy paste error. Thanks!